server/internal/infrastructure/live-store/inmemory/tree_signing_session.go

```diff
@@ -76,7 +82,12 @@ func (s *treeSigningSessionsStore) AddNonces(
 	s.sessions[roundId].Nonces[pubkey] = nonces
 
 	if len(s.sessions[roundId].Nonces) == s.sessions[roundId].NbCosigners-1 {
-		s.nonceCollectedCh[roundId] <- struct{}{}
+		if ch, exists := s.nonceCollectedCh[roundId]; exists {
```

Comment:
let's return error if the channel doesn't exist because it means something bad happened! we don't want to ignore such situation if we have this kind of check. The existing code assumed something bad can't just happen instead